### PR TITLE
Fix some smaller problems with respect to types and `alleinerz`

### PR DESCRIPTION
**Pull request by ChristianZimpelmann**
### Todo

- [x] Set type of `wohnfläche_hh` to ``float``.
- [x] Make sure data created with `synthetic_data` has types expected by GETTSIM.
- [x] Adjust calculation for Unterhaltsvorschuss (and respective tests) such that only the single parent is defined as 'alleinerz' and not the children. (this is related to #270)

### CODE REVIEW

File: gettsim/tests/test_data/unterhalt.csv [hunk 1]
```diff
@@ -1,8 +1,8 @@
 hh_id,tu_id,p_id,alleinerz,alter,bruttolohn_m,eink_selbst_m,sonstig_eink_m,arbeitsl_geld_m,sum_ges_rente_priv_rente_m,kapitaleink_brutto_m,eink_vermietung_m,jahr,monat,,,unterhaltsvors_m,,,,,These test cases wer just to check that we return a Series of nans. We exclude this test case for now.,,,,,,,,,,,,,
 5,5,6,TRUE,45,0,0,0,0,0,0,0,2018,1,,,0,,,,,1,1,23,FALSE,40,0,0,0,0,0,0,0,FALSE,2017
-5,5,7,TRUE,10,0,0,0,0,0,0,0,2018,1,,,205,,,,,1,1,25,FALSE,35,0,0,0,0,0,0,0,FALSE,2017
+5,5,7,FALSE,10,0,0,0,0,0,0,0,2018,1,,,205,,,,,1,1,25,FALSE,35,0,0,0,0,0,0,0,FALSE,2017
```
Comment (LauraGergeleit) [inline gettsim/tests/test_data/unterhalt.csv:3]: There should be other tests for which `alleinerz` for children is still evaluated as `True`, for example in `arbeitsl_geld_2.csv`

**Review by LauraGergeleit — COMMENTED** [review]
Looks good! One comment with respect to the tests...

**Comment by hmgaudecker** [comment]
Random thought while browsing issues: Is it useful to add a check as proposed in #402 here? While  you are looking at those testcases anyhow...

**Comment by ChristianZimpelmann** [comment]
> Random thought while browsing issues: Is it useful to add a check as proposed in https://github.com/iza-institute-of-labor-economics/gettsim/issues/402 here? While you are looking at those testcases anyhow...

Instead of adding it here, I would like to put some deeper thought into how to specify these kind of checks next year (see my new comment in #402)

**Review by LauraGergeleit — APPROVED** [review]
Looks good!

**Pull request merged**